src/utils/components/SSHAccess/components/SSHCommand.tsx

```diff
@@ -38,22 +35,37 @@ const SSHCommand: React.FC<SSHCommandProps> = ({
   sshServiceLoaded,
 }) => {
   const { t } = useKubevirtTranslation();
-  const [sshService, setSSHService] = useState<IoK8sApiCoreV1Service>();
-  const { command, user, sshServiceRunning } = useSSHCommand(vm, sshService);
+  const [sshService, setSSHService] = useState<IoK8sApiCoreV1Service | null>();
+  const { command, sshServiceRunning } = useSSHCommand(vm, sshService);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<Error>();
 
-  const onSSHChange = (enableSSH: boolean) => {
+  const onSSHChange = async (newServiceType: SERVICE_TYPES) => {
     setLoading(true);
 
-    const request = enableSSH
-      ? createSSHService(vm, vmi).then(setSSHService)
-      : deleteSSHService(sshService).then(() => setSSHService(undefined));
+    try {
+      const deletePromise = sshService
```

Comment:
why u need to use empty promise? what happen if u return undefined Promise.allSettled don't like it?